### PR TITLE
D45 doc update

### PR DESCRIPTION
**Pull request by d45**
This is a placeholder pull request. It has some of the changes we need for the 2018.2 release. 
Feel free to add to it. @johnDance @jz-huang @Kovner 

 Changes in this update:
* Added Known issues topic (linked from landing page and TOC).
* Add XSS page (rough draft)
* Reorganized TOC slightly, added a Security folder to group related toipcs. 

To do: 
* Review Known Issues
* Review and extend XSS topic
* Add to the Release Notes for this release (links to Tableau, JS library)

### CODE REVIEW

File: docs/trex_create.md [hunk 1]
```diff
@@ -28,20 +28,20 @@ These instructions assume that you already have cloned or download the Extension
...
-        <?xml version="1.0" encoding="utf-8"?> 
+        <?xml version="1.0" encoding="utf-8"?>
         <manifest manifest-version="0.1" xmlns="http://www.tableau.com/xml/extension_manifest">
           <dashboard-extension id="com.example.extensions.name" extension-version="0.1.0">
             <default-locale>en_US</default-locale>
             <name resource-id="name"/>
             <description>Extension Description</description>
-            <author name="USERNAME" email="USER@example.com" organization="My Company" website="www.example.com"/>
+            <author name="USERNAME" email="USER@example.com" organization="MyCo" website="https://www.example.com"/>
             <min-api-version>0.8</min-api-version>
```
Comment (johnDance) [inline docs/trex_create.md]: Should we change this to 1.0?

File: docs/trex_debugging.md [hunk 2]
```diff
@@ -82,3 +82,6 @@ Due to changes in the start-up sequence, you can no longer enable remote debuggi
...
 
+#### Extensions and Internet Explorer 11
+The Extensions API library uses modern JavaScript constructs, such as promises, arrow functions, typed arrays, and iterators. The library was built for ECMAScript 5 compatibility, and provides the appropriate polyfills. However, if you use ECMAScript 6 language constructs, your code will have syntax errors in Internet Explorer 11 (IE11). If you want your extension to run in IE11, you should avoid using ECMAScript 6 features and language constructs (see http://es6-features.org/) or have the appropriate polyfills and tools to convert to ECMAScript 5. If you are using Typescript, for example, TypeScript allows you to use ECMAScript 6 language constructs, but target ECMAScript 5 compatibility.
```
Comment (johnDance) [inline docs/trex_debugging.md]: Didn't Lee think this was too much information? I don't know what he recommended instead.

File: docs/trex_release-notes.md [hunk 3]
```diff
@@ -8,12 +8,18 @@ layout: docs
...
+Coming soon.
+ 
+
 ### Tableau 2018.2 (Beta 2)
 *June 4, 2018*
 
 Update of the Tableau Extensions API
 
-- Tableau Extensions API library: tableau-extensions-0.16.0.js
+- Tableau Extensions API library: `tableau-extensions-0.16.0.js`
```
Comment (johnDance) [inline docs/trex_release-notes.md]: We are moving to version 1.0.0 this coming week.

File: docs/trex_release-notes.md [hunk 4]
```diff
@@ -8,12 +8,18 @@ layout: docs
 * TOC
 {:toc}
 
+### Tableau 2018.2 (Release Candidate)
```
Comment (johnDance) [inline docs/trex_release-notes.md]: Shouldn't this be Tableau 2018.2
The (Release Candidate) seems like internal information.

File: docs/trex_xss_guidance.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+If your dashboard extension accepts text input from users through forms or text boxes, you will want to ensure that you aren't accidentally introducing vulnerabilities. The best prevention for XSS attacks is to validate and encode any user input. There is a lot of good information available on the web for developers about how to protect web applications from attacks. For example, see  [Cross Site Scripting Prevention Cheat Sheet](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet){:target="_blank"} from the Open Web Application Security Project (OWASP).
+
+#### Key points:
+
+* Always validate user-provided text input into your HTML pages.
+  If your extension uses forms or text boxes to collect user input, be sure to validate and verify that the input conforms to what you expect. Use regular expressions to verify the input. 
+* Use HTML entity encoding and escape the characters `&`, `<`, `>`, `"`, `'`, and `/`.  For example, use entity names or numbers `&gt;` or `&#62;` for the greater than character `>`.
+* If your extension makes use URIs as input, use JavaScript methods, such as `encodeURI` and `encodeURIComponent` for encoding.
```
Comment (johnDance) [inline docs/trex_xss_guidance.md]: "If your extension makes use _of_ URIs" (You are missing "of")

File: docs/trex_create.md (hunk 1 above)
Comment (d45) [reply docs/trex_create.md]: Yes. I'll change that.

File: docs/trex_debugging.md (hunk 2 above)
Comment (d45) [reply docs/trex_debugging.md]: @LGraber to see if he has any comments about what to leave out.  I sent this out for review a month ago but didn't hear back then.

File: docs/trex_release-notes.md (hunk 3 above)
Comment (d45) [reply docs/trex_release-notes.md]: Yes. The 2018.2 release will say 1.0.   We just put the latest release on the top of the page. 
The release notes for Beta 2 on June 4 point to the 0.16 library and Tableau Beta 2 downloads.

File: docs/trex_release-notes.md (hunk 4 above)
Comment (d45) [reply docs/trex_release-notes.md]: Yes. Good point. I changed that to just 2018.2.

File: docs/trex_xss_guidance.md (hunk 5 above)
Comment (d45) [reply docs/trex_xss_guidance.md]: Thanks!

**Pull request merged**